chain/jsonrpc/tests/rpc_query.rs

```diff
@@ -552,3 +552,112 @@ fn test_gas_price() {
         assert!(gas_price.gas_price > 0);
     });
 }
+
+#[test]
+#[ignore] // https://github.com/nearprotocol/nearcore/issues/2789
+fn test_query_view_account_non_existing_account_must_return_error() {
+    test_with_client!(test_utils::NodeType::NonValidator, client, async move {
+        let query_response = client
+            .query(RpcQueryRequest {
+                block_id_or_finality: BlockIdOrFinality::latest(),
+                request: QueryRequest::ViewAccount { account_id: "invalidaccount".to_string() },
+            })
+            .await
+            .unwrap();
+        if let QueryResponseKind::ViewAccount(_) = query_response.kind {
+            panic!("queried view account for not exsiting account, but received success instead of error");
+        }
+        assert!(true);
```

Comment:
```suggestion
        assert!(!matches!(query_response.kind, QueryResponseKind::ViewAccount(_)), "queried view account for not exsiting account, but received success instead of error");
```